numpy/linalg/linalg.py

```diff
@@ -2039,28 +2041,35 @@ def lstsq(a, b, rcond="warn"):
                                  0, work, lwork, iwork, 0)
     if results['info'] > 0:
         raise LinAlgError('SVD did not converge in Linear Least Squares')
-    resids = array([], result_real_t)
-    if is_1d:
-        x = array(ravel(bstar)[:n], dtype=result_t, copy=True)
-        if results['rank'] == n and m > n:
-            if isComplexType(t):
-                resids = array([sum(abs(ravel(bstar)[n:])**2)],
-                               dtype=result_real_t)
-            else:
-                resids = array([sum((ravel(bstar)[n:])**2)],
-                               dtype=result_real_t)
+
+    # undo transpose imposed by fortran-order arrays
+    b_out = bstar.T
+
+    # b_out contains both the solution and the components of the residuals
+    x       = b_out[:n,:]
+    r_parts = b_out[n:,:]
+    if isComplexType(t):
+        resids = sum(abs(r_parts)**2, axis=-2)
```

Comment:
I wish we had a sensible `power` or so, but to avoid a needless square root, one can do 
`sum(r_parts.real**2 + r_parts.imag**2, axis=-2)`

Reply:
`r_parts * r_parts.conj()` is probably a little faster. I'd rather leave this untouched though for now, since that would probably change results by a ULP.

Reply:
It's not (at least on my machine), but fine to let this be.

Reply:
It's not faster, or doesn't introduce the ULP error. Seems to me that there must be some value for which `abs(x)**2 != x * x.conj()`. Of course, the `x * x.conj()` value is closer to the true result.

Reply:
I just meant that `x*x.conj()` is slower than `x.real**2 + x.imag**2` (which makes sense, as the former does a few useless multiplications that cancel). I do agree that there must be values of `abs(x)**2` that are slightly less correct, given the sqrt and square after calculating `x.real**2+x.imag**2`

Anyway, fine to not worry about it here!

Reply:
I've contemplated adding a ufunc for the squared absolute value, the main problem seems to be the name.